docs/Encoder.md

```diff
@@ -1,57 +1,103 @@
-| Input voltage (A, A̅, A, B̅, Z or Z̅)           | -11 V            | 14 V     |
-| Differential input voltage      | -14 V            | 14 V    |
-| Output voltage (OUT A, OUT B, OUT Z) | -0.5 V              | (Vcc + 0.5) V  |
+The AMDC REV D hardware supports two independent encoder interfaces. These must be incremental encoders outputting differential ABZ signals (i.e. the encoder outputs both the non-inverting and inverting signal, e.g. A and A̅). If the encoder only provides single-ended ABZ outputs, an adapter must be used between the encoder and the AMDC which converts the single-ended outputs to the differential signals required by the AMDC. For proper operation, the encoder ground should be connected to the AMDC ground via a pin on the DB9 connector. For powering the encoder, the user has two options: the AMDC provides 5V power via the DB9 connector, or the user can use an external power supply.
 
-Exposure to absolute maximum ratings for extended period of time may affect device reliability. Therefore, it is recommended to operate under recommended operating conditions:
+#### Maximum Encoder Frequency Output
 
-| Parameter | MIN            | NOM                | MAX|
-|-------------------|-----------------|----------------------|----------------------|
-| Supply voltage Vcc             | 4.5 V | 5 V          | 5.5 V |
-| High-level input voltage (logic HIGH) | 2 V |      | Vcc     |
-| Low-level input voltage (logic LOW)   | 0 V |      | 0.8 V    |
-| Common-mode input voltage | -7 V      |      | 7 V  |
+Users must select an appropriate encoder for their application. Typically, the decision is based on cost, physical size, rotor speed, and maximum supported frequency output. The last metric, maximum frequency output, is determined by the AMDC hardware.
 
-When chip is operated at nominal conditions (Vcc = 5 V, room temperature 25<sup>0</sup>), the output signals of the chip are such that the high-level output voltage is 3.8 V or more and low-level output voltage is around 0.2 V. Typical current the chip consumes is about 10 mA. This translates to about 50 mW of power consumption with 5 V supply. The rise and fall times of the output signal are both approximately equal to 4 ns. Assuming the system as first order, the bandwidth of the chip would approximately be equal to 0.35/4 ns = 87.5 MHz.
+The maximum encoder signal frequency that the AMDC is able to receive depends on the FPGA clock frequency, which is nominally 200MHz.  It is suggested that there is a minimum of a 10x oversampling ratio. This means the frequency of the encoder square wave output should be less than 20MHz (20 million pulses per second).
 
-Before connecting encoder to AMDC, it should be checked whether encoder output signals satisfy recommended operating conditions of the chip. More detailed information on the operating conditions can be found in the [datasheet](http://www.ti.com/lit/ds/symlink/am26c32.pdf).
+##### Encoder Example Application

```

Comment:
This example is great! Very nice idea to include it.

Is it based around an actual encoder product that is commercially available? If so, can we provide a link to the product? If not, could we update this to correspond to an actual encoder (and provide a link to the product). This would present more a complete example that I think would be helpful to users to see.

Reply:
@Anvar-Khamitov @elsevers I don't think this is a real encoder... I have used US Digital encoders with okay luck in the past: https://www.usdigital.com/products/encoders/incremental/kit/E5

On their website, you can specify the `CPR` you want. NOTE: This is **not** the definition we are using in these docs! CPR for US Digital means "cycles per rev" where CPR * 4 = PPR. In this doc, we are calling counts per rev, "CPR," to mean PPR from US Digital. Very confusing, yes. We care about the larger number (the 4x number)!

The example in the doc was written with easy numbers and easy math. Hopefully we can find a real product that also has easy numbers for easy math.

Reply:
Sounds good. Instead of modifying the example to pertain to an exact encoder, we could instead link to this encoder at the end of the example and just explain how to interpret its terminology for the example calculations.